indico/modules/categories/client/js/calendar.js

```diff
@@ -100,14 +217,24 @@ import {$T} from 'indico/utils/i18n';
     });
     calendar.render();
 
-    $('.js-calendar-datepicker').datepicker({
-      dateFormat: 'yy-mm-dd',
-      beforeShowDay(date) {
-        return [date.getDate() === 1, ''];
-      },
-      onSelect(date) {
-        calendar.gotoDate(date);
-      },
-    });
+    // set up the datepicker
+    const elem = $('.fc-toolbar-chunk:eq(2) button:first')
+      .css('background-color', 'transparent')
+      .css('border-color', 'transparent')
+      .css('background', 'transparent')
+      .get(0);
+    ReactDOM.render(
+      <SingleDatePicker
+        onDateChange={date => calendar.gotoDate(date.toDate())}
+        placeholder={Translate.string('Go to date...')}
```

Comment:
It's *really* weird that this thing looks like an input field and not like a button

![image](https://github.com/indico/indico/assets/179599/eae7eb36-01e7-48ca-aa30-af65790d0e91)

Reply:
So, this is something kinda hard to explain... it seems the react-dates component ONLY can render itself with an input. On the other hand... fullcalendar.js ONLY allows to customize it by using buttons. So, what I did is mixing both: there's a rendered button that I managed to hide with CSS and at the same time I used the only component that the current datepicker is allowed to use.

I'm aware this is a dirty hack, but it's the best approach I managed to get. If we'd like to dive deeper we'd have to use another react component that allows more flexibility.

Reply:
I don't think so. Take a look at this: https://react-dates.github.io/react-dates/?path=/story/singledatepicker-sdp--default

There's no way to render the component without an input field.

Similar for fullcalendar's custom buttons: https://fullcalendar.io/docs/customButtons

Reply:
https://react-dates.github.io/react-dates/?path=/story/daypicker--single-month